nepalingo-web/src/components/StreakContext.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  ReactNode,
+} from "react";
+import supabase from "./userAuth/supabaseClient";
+import { useAuth } from "./userAuth/AuthContext";
+
+interface StreakContextProps {
+  currentStreak: number;
+  longestStreak: number;
+  updateStreak: () => void;
+}
+
+export const StreakContext = createContext<StreakContextProps>({
+  currentStreak: 0,
+  longestStreak: 0,
+  updateStreak: () => {},
+});
+
+export const StreakProvider = ({ children }: { children: ReactNode }) => {
+  const { user } = useAuth();
+  const [currentStreak, setCurrentStreak] = useState(1);
+  const [longestStreak, setLongestStreak] = useState(1);
+
+  useEffect(() => {
+    const fetchStreakData = async () => {
+      if (user) {
+        const { data } = await supabase
+          .from("user_daily_streaks")
+          .select("*")
+          .eq("user_id", user.id)
+          .single();
+
+        if (data) {
+          setCurrentStreak(data.current_streak);
+          setLongestStreak(data.longest_streak);
+        }
+      }
+    };
+
+    fetchStreakData();
+  }, [user]);
+
+  const updateStreak = async () => {
+    const currentDate = new Date().toISOString().split("T")[0]; // Get current date
+    const lastUpdateDate = localStorage.getItem("lastUpdateDate");
```

Comment:
I'm confused where we're setting the localStorage value, did you forget to push up some code?
Also - we should be relying on the database (supabase) for that information not localstorage

Reply:
since the lastUpdatedDate doesn't change frequently, I thought it would be efficient to store it in localstorage. It would reduce some queries. well...I will change it to database, no worries!

Reply:
there isn't a table for lastUpdatedDate. the variable keeps track of the date, so the user doesn't update streaks everytime he gets to home page, until the dates have changed. maybe I can directly use the state lastUpdateDate to keep track of the last date the streak was updated.

Reply:
![CleanShot 2024-07-17 at 23 25 51@2x](https://github.com/user-attachments/assets/92ebc7c5-caf9-4e84-8eb0-bcd582539f22)
I think `lastUpdateDate` would either be `streak_started_date` or `streak_ended_date`

Reply:
hmmmm.....makes sense